src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,247 @@
+import sys
+import logging
+import subprocess
+import pkg_resources
+import json
+import pathlib
+
+import sas
+
+
+
+class PackageGatherer:
+    """
+    A class used to gather packages/modules  used by SasView and their current installed version
+
+    Methods
+    -------
+    log_installed_modules
+        Log version number of locally installed python packages
+
+    log_imported_modules
+        Log version number of python packages imported in this instance of SasView.
+
+    get_imported_modules
+        Get a dictionary of imported module version numbers
+
+    remove_duplicate_modules
+        Strip duplicate instances of each module
+
+    format_unattainable_modules_list
+        Format module names in the unattainable_modules list
+    """
+
+
+    def log_installed_modules(self):
+        """ Log version number of locally installed python packages
+
+        Use pip list to create a dictionary of installed modules as the keys, with their respective version numbers
+        as the values. Only packages available through pip will be included.
+
+        Returns
+        -------
+        None
+        """
+        installed_modules = {'python': sys.version}
+
+        # Get python modules installed locally
+        installed_modules_json = json.loads(subprocess.check_output("pip list -l --format=json"))
+        for mod in installed_modules_json:
+            installed_modules[mod['name']] = mod['version']
+
+        logging.info(f"Installed modules\n"
+                     f"{installed_modules}")
```

Comment:
It would be polite to sort these, and maybe format them a little nicer as:
```python
logging.info("Installed modules:\n  " + "\n".join(f"{k}: {v}" for k, v in sorted(installed_modules.items())))
```
If log level is warning or error then this message won't show. This will cause user confusion if for some reason we lower the log level but leave a menu item active that appears to do nothing. Do we have a mechanism other than logging for dumping things to the SasView console?

Reply:
I see what you’re saying, and I get that it may be confusing to users if the logging level is set to say error.

But isn’t the primary use of logging to debug, and this tool is primarily a debugging tool, so I would argue that using logging is the most consistent way to show this information. However it may be useful to add something that will always show in the console, like “modules successfully logged”

Also maybe set the logging level to debug in the __init__ section of the class, since this is a debugging tool?

Reply:
Okay, not worth worrying about log level.